src/Service/ConsoleOptionsReader.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the box project.
+ *
+ * (c) Anatoliy Melnikov <5785276@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Aeliot\PhpCsFixerBaseline\Service;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\Finder;
+
+class ConsoleOptionsReader
+{
+    private string $rootDirectory;
+
+    /** @var array<string, string>  */
+    private array $option;
+
+    private static ?self $instance = null;
+
+    public function __construct()
+    {
+        self::$instance = $this;
+        $this->option = getopt('ab:c:d:f:w:', ['absolute', 'baseline:', 'config:', 'dir:', 'finder:', 'workdir:']);
+        $this->rootDirectory = ($this->option['d'] ?? $this->option['dir']) ?? '';
+
+        var_dump($this->rootDirectory);
+    }
+
+    public static function getInstance(): self
+    {
+        return self::$instance ?: throw new \LogicException('No instance provided');
+    }
+
+    public function getAsArray(): array
+    {
+        return [
+            'baselinePath' => $this->getBaselinePath(),
+            'config' => $this->getConfig(),
+            'finder' => $this->getFinder(),
+            'relative' => $this->getRelative(),
+            'workdir' => $this->getWorkdir(),
+        ];
+    }
+
+    private function getBaselinePath(): string
+    {
+        return $this->getAbsolutePath($this->getOptionValue('b', 'baseline', '.php-cs-fixer-baseline.json'));
+    }
+
+    private function getConfig(): Config
+    {
+        $configPath = $this->getAbsolutePath($this->getOptionValue('c', 'config', '.php-cs-fixer.dist.php'));
+
+        return require $configPath;
+    }
+
+    private function getFinder(): Finder
```

Comment:
Return data type declared `Finder` by returns `string`

Reply:
nope)
![image](https://github.com/user-attachments/assets/0effdd1e-67e0-458a-8e06-da9aa33b65d1)

Reply:
OMG! I missed it. Thank you